CONTRIBUTING.md

```diff
@@ -1,6 +1,132 @@
+
+## Contribution Guidelines
+
+---
+
+Contributions from everyone, experienced and inexperienced, are welcome! If you don't know where to start, look at the [Open Issues](https://github.com/StingraySoftware/stingray/issues) and/or get involved in our [Slack channel](http://slack-invite.timelabtechnologies.com/) . This code is written in Python 3.5+. Tests run at each commit during Pull Requests, so it is easy to single out points in the code that break this compatibility.
+
+- **Branches:**
+    - Don't use your master ****branch(forked) for anything. Consider deleting your master ****branch.
+    - Make a new branch, called a feature branch, for each separable set of changes: “one task, one branch”.
+    - Start that new feature branch from the most current development version of stingray.
+    - Name of branch should be the purpose of change eg. *bugfix-for-issue20* or *refactor-lightcurve-code.*
+    - Never merge changes from stingray/master into your feature branch. If changes in the development version require changes to our code you can rebase, but only if asked.
+- **Commits:**
+    - Make frequent commits.
+    - One commit per logical change in the code-base.
+    - Add commit message.
+- **Naming Conventions:**
+    - Change name of the remote origin(*yourusername/stingray*) to your *github-username.*
+    - Name the remote that is the primary stingray repository( *StingraySoftware/stingray*) as  stingray.
+
+### Contribution Workflow
+
+These, conceptually, are the steps you will follow in contributing to stingray. These steps keep work well organized, with readable history. This in turn makes it easier for project maintainers (that might be you) to see what you’ve done, and why you did it:
+
+1. Regularly fetch latest stingray development version `stingray/master` from GitHub.
+2. Make a new feature branch. **Recommended:** Use virtual environments to work on branch.
+3. Editing Workflow:
+    1. One commit per logical changes.
+    2. Run tests to make sure that changes don't break existing code.
+    3. Code should have appropriate docstring.
+    4. Update appropriate documentation if necessary and test it on sphinx.
+    5. Write a test for code changes.
+    6. If  modifications require more than one commit, break changes into smaller commits.
+    7. Push the code on your remote(forked) repository.
+4. All code changes should be submitted via PR's(i.e. fork, branch, work on stuff, just submit pull request).
+Code Reviews are super-useful: another contributor can review the code, which means both the contributor and reviewer will be up to date with how everything fits together, and can get better by reading each other's code! :)
+5. Take feedback and make changes/revise the PR as asked.
+6. Don't merge using web interface if your branch falls behind master. Fetch and rebase.
```

Comment:
This is probably inapplicable, because there is a very small number of people who can merge.

Reply:
Commented out